sycl/source/detail/jit_compiler.cpp

```diff
@@ -725,6 +718,14 @@ jit_compiler::fuseKernels(QueueImplPtr Queue,
     ++KernelIndex;
   }
 
+  if (const auto Ranges = gatherRanges(InputKernelInfo);
```

Comment:
Why gather the ranges here and not collect them during the loop above?

Reply:
Done